core/lib/zksync_core/src/commitment_generator/mod.rs

```diff
@@ -210,11 +214,35 @@ impl CommitmentGenerator {
             }
             state_diffs.sort_unstable_by_key(|rec| (rec.address, rec.key));
 
+            let kzg_settings = load_kzg_settings();
```

Comment:
I think you don't need to load KZG settings for pre-1.4.2, so you probably should move this logic to the `if` body (this way for non-1.4.2 the corresponding `once_cell` will never be instantiated).